server/controllers/notification.js

```diff
@@ -0,0 +1,89 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /notifications/
+// @desc create a new notification
+// @access Public
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const { type, title, description, receiver } = req.body;
+  
+  if (!type || !title || !description || !receiver){
+    res.status(400);
+    throw new Error("Bad request! Missing type, title, description or receiver!");
+  }
+
+  const notification = await Notification.create({
+    sender: req.user.id,
+    receiver,
+    type,
+    title,
+    description,
+  });
+
+  res.status(200).json({
+    success: {
+      notification: notification,
+    },
+  });
+});
+
+// @route PATCH /notifications/:notificationId/mark-read/
+// @desc changes mark as read property to true on a notification
+// @access Private
+
+exports.markAsRead = asyncHandler(async (req, res, next) => {
+  const { notificationId } = req.params;
+  const foundNotification = await Notification.findById(notificationId);
+
+  if (foundNotification.user.toString() !== req.receiver.id) {
```

Comment:
```suggestion
  if (foundNotification.receiver.toString() !== req.user.id) {
```
We need to compare if the logged in user is the receiver of a notification here :)

Reply:
Okay I switched it around. I think its good to merge now?